Documentation/gettingstarted/encryption-ipsec.rst

```diff
@@ -0,0 +1,288 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _encryption_ipsec:
+
+****************************
+IPsec Transparent Encryption
+****************************
+
+This guide explains how to configure Cilium to use IPsec based transparent
+encryption using Kubernetes secrets to distribute the IPsec keys. After this
+configuration is complete all traffic between Cilium-managed endpoints, as well
+as Cilium-managed host traffic, will be encrypted using IPsec. This guide uses
+Kubernetes secrets to distribute keys. Alternatively, keys may be manually
+distributed, but that is not shown here.
+
+Packets destined to the same node they were sent out of are not encrypted.
+This is a intended behavior as it doesn't provide any benefits because the
+raw traffic on the node can be seen.
+
+Transparent encryption is not currently supported when chaining Cilium on top
+of other CNI plugins. For more information, see
+`GitHub issue #15596 <https://github.com/cilium/cilium/issues/15596>`_.
+
+Generate & import the PSK
+=========================
+
+First, create a Kubernetes secret for the IPsec configuration to be stored. The
+example below demonstrates generation of the necessary IPsec configuration
+which will be distributed as a Kubernetes secret called ``cilium-ipsec-keys``.
+A Kubernetes secret should consist of one key-value pair where the key is the
+name of the file to be mounted as a volume in cilium-agent pods, and the
+value is an IPSec configuration in the following format:
+
+``key-id encryption-algorithms PSK-in-hex-format key-size``
+
+.. note::
+
+    ``Secret`` resources need to be deployed in the same namespace as Cilium!
+    In our example, we use ``kube-system``.
+
+In the example below we use GMC-128-AES, but
+any of the supported Linux algorithms may be used. To generate, use the
+following:
+
+.. parsed-literal::
+
+    $ kubectl create -n kube-system secret generic cilium-ipsec-keys \\
+        --from-literal=keys="3 rfc4106(gcm(aes)) $(echo $(dd if=/dev/urandom count=20 bs=1 2> /dev/null| xxd -p -c 64)) 128"
+
+The secret can be seen with ``kubectl -n kube-system get secret`` and will be
+listed as "cilium-ipsec-keys".
+
+.. parsed-literal::
```

Comment:
Before we merge, can you please check that this block is rendered correctly? I know you didn't touch it other than the renaming, but I wonder if the lack of empty line between directive name and block content might result in some error. Might be a good occasion to fix if that were the case.

Reply:
Thanks for the review! It looks like it's rendered correctly to me (from the preview link in PR description): https://nervous-ptolemy-1de0e7.netlify.app/gettingstarted/encryption-ipsec.html#generate-import-the-psk

Reply:
Ok my bad then, thanks for checking!